apps/twap/src/app/utils/processConditionalOrders.ts

```diff
@@ -0,0 +1,154 @@
+import { providers } from 'ethers';
+import { getConditionalOrderId } from '../utils/getConditionalOrderId';
+import { getOrderStatus } from '../utils/getOrderStatus';
+import { parseOrderStruct } from '../utils/parseOrderStruct';
+import { generateOrderParts } from '../utils/generateOrderParts';
+import { Wallet } from '../data/wallet';
+import { Order } from '../data/order';
+import { OrderPart } from '../data/orderPart';
+import { Block } from '../data/block';
+import { getSingleOrder } from './getSingleOrder';
+import { FastifyInstance } from 'fastify';
+import { ComposableCoW__factory } from '@cow-web-services/abis';
+import { getBlockInfo } from './getBlockInfo';
+export const COMPOSABLE_COW_ADDRESS =
+  '0xF487887DA5a4b4e3eC114FDAd97dc0F785d72738';
+
+export async function processConditionalOrders(
```

Comment:
in general, i see a lot of utils that are more doing some bussiness logic than being a util.

Also, is see bussines logic mixed with `fastify`.

However, since you were consistent and for now the project is small, you can ignore this comment. If this is fastify way of doing things, let me know!

Reply:
Good point, I'll categorise these as handlers in that case; since they are not utils in the strictest sense of the word indeed.

As far as the fastify instance goes, that's essentially poor man's dependency injection at the moment. Missing part here most likely is that this should be a plugin as well (fastify really likes plugins), so that I can have fastify instance accessible. Will update it tomorrow as part of a bit of a refactor here.

Reply:
Updated the structure. I've essentially;
* Created a new folder `conditionalOrders`. This would be in line with nodejs best practices (domain based folder structures, as outlined at https://github.com/goldbergyoni/nodebestpractices#-11-structure-your-solution-by-business-components)
* Moved logic that is not a util to `conditionalOrders`.
* Moved `processConditionalOrders` to plugins as a new plugin. Also created `blockInfo` plugin to simplify, and added a new function `getConditionalOrderCreatedEvents` under `conditionalOrders`.